### ISSUE TITLE
Possible enum_glob_use false positive

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#8318]
### Summary

This code:

```rust
#![warn(clippy::pedantic)]

fn main() {
    enum Ops {Add, Sub, Mul, Div};
    use Ops::*;
    
    let op = Add;
    let s = match op {
        Add => "+",
        Sub => "-",
        Mul => "*",
        Div => "/",
    };
    println!("{}", s);
}
```

Gives:

```rust
warning: usage of wildcard import for enum variants
 --> src/main.rs:5:9
  |
5 |     use Ops::*;
  |         ^^^^^^ help: try: `Ops::{Add, Div, Mul, Sub}`
  |
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![warn(clippy::pedantic)]
  |         ^^^^^^^^^^^^^^^^
  = note: `#[warn(clippy::enum_glob_use)]` implied by `#[warn(clippy::pedantic)]`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_glob_use
```

In this case glob use seems OK to me...

### Reproducer

I tried this code:

```rust
<code>
```

I saw this happen:

```
<output>
```

I expected to see this happen:
